src/shogun/machine/Pipeline.cpp

```diff
@@ -134,9 +134,17 @@ namespace shogun
 			else
 			{
 				auto machine = shogun::get<std::shared_ptr<Machine>>(stage.second);
-				if (machine->train_require_labels())
-					machine->set_labels(m_labels);
-				machine->train(current_data);
+				try
```

Comment:
why do you need this?

Reply:
the `linear machine` now switch `train(shared_ptr<Features> data)` to `train(const shared_ptr<Features>& data, const std::shared_ptr<Labels>& labs)`, but other `machine`s still use `train(shared_ptr<Features> data)`, i can't find a good way to detect which `train` machine objects have.

Reply:
what about adding wrappers?

Reply:
hmm I don't understand, right now `train(const std::shared_ptr<Features>& data, const std::shared_ptr<Labels>& lab)` calls `train(std::shared_ptr<Features> data)` no? So why not always call `machine->train(current_data, m_labels);`?

Reply:
oh i see you changed it in this PR. So is this `try` temporary until you finished refactoring all the machines?